src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -2156,7 +2156,7 @@ enum Nf {
 
 #define INSN(NAME, funct3, op)                                                               \
   void NAME(int32_t offset) {                                                                \
-    assert_cond(is_imm_in_range(offset, 11, 1));                                             \
+    assert(is_simm12(offset) && ((offset % 2) == 0), "just checking");                       \
```

Comment:
Could you please improve the assert message? like `invalid imm` or something more specific.

Reply:
Thanks! I will use `invalid encoding` instead.